### DISCUSSION TITLE
Is there a way to retrieve the generation parameters of a specific image from its EXIF data?

### DISCUSSION THREAD

**Discussion by zanllp** [Comfy-Org/ComfyUI#1007] [Q&A] [answered]
I am the author of [sd-webui-infinite-image-browsing](https://github.com/zanllp/sd-webui-infinite-image-browsing), and someone has requested that my image browser support ComfyUI. I have recently started working on this feature, but I have encountered some difficulties, mainly in figuring out how to correctly extract the generation parameters from the image's EXIF data.

If there is only one image, it's relatively simple, and I can just use the key "3" to look it up as an entry point.

![image](https://github.com/comfyanonymous/ComfyUI/assets/25872019/96a2e05c-8482-47ee-afac-7a29b8ae78e9)

However, for more complex images like the one shown below, it's much more difficult because I don't know which sampler was used as the entry point for the current image.

![image](https://github.com/comfyanonymous/ComfyUI/assets/25872019/5508847e-d92b-442c-8fdd-316a7722c2c7)

Related issue: https://github.com/zanllp/sd-webui-infinite-image-browsing/issues/202. You can find the original image with EXIF data in that issue.

**Comment by ltdrdata** [comment, marked as answer]
This issue is related to https://github.com/comfyanonymous/ComfyUI/issues/612

As of now, there is no simple solution. The issue is under discussion, but it requires further development.

**Reply by zanllp** [reply]
Alright, thank you for the quick response

**Comment by wcurti** [comment]
I've resorted to a kludgy solution that rewrites the Save Image node to save metadata in a readable format, then converting every single value I want to save from the starting nodes to an input and arranging a series of primitive nodes to set up my initial values and connecting them to both the starting node and save node inputs at once. 

I hate it but it works.
